### ISSUE TITLE
4.x: archetype - module-info opt-in

### ISSUE THREAD

**Issue by romain-grecourt** [helidon-io/helidon#7735]
## Environment Details
* Helidon Version: 4.0.0-RC1

## Problem Description

Create a new option that conditionally adds a module-info and a module main class(mp only).

This option should go under the `Package and Deployment` step:

```properties
id=jpms
name=Module support (JPMS)
description=Add a module-info
```

When flavor=mp, the description should be:

```properties
description=Add a module-info and a module main class
```

Add `jpms=false` as a preset to all app-types (other than custom).

This needs to be back-ported to 3.x as well.

**Comment by barchetta** [comment]
See also #7722

**Comment by tvallin** [comment]
Closed by #7775
